api/dash0monitoring/v1alpha1/operator_types.go

```diff
@@ -0,0 +1,173 @@
+// SPDX-FileCopyrightText: Copyright 2024 Dash0 Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package v1alpha1
+
+import (
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	common "github.com/dash0hq/dash0-operator/api/dash0monitoring"
+)
+
+const (
+	OperatorFinalizerId = "operator.dash0monitoring.com/dash0monitoring-operator-finalizer"
+)
+
+// Dash0OperatorConfigurationSpec defines the desired state of the Dash0 monitoring resource.
+type Dash0OperatorConfigurationSpec struct {
+	// The URL of the observability backend to which telemetry data will be sent. This property is mandatory. The value
+	// needs to be the OTLP/gRPC endpoint of your Dash0 organization. The correct OTLP/gRPC endpoint can be copied fom
+	// https://app.dash0.com/settings. The correct endpoint value will always start with `ingress.` and end in
+	// `dash0monitoring.com:4317`.
+	//
+	// +kubebuilder:validation:Mandatory
+	IngressEndpoint string `json:"ingressEndpoint"`
+
+	// The Dash0 authorization token. This property is optional, but either this property or the SecretRef property has
+	// to be provided. If both are provided, the AuthorizationToken will be used and SecretRef will be ignored. The
+	// authorization token for your Dash0 organization can be copied from https://app.dash0.com/settings.
+	//
+	// +kubebuilder:validation:Optional
+	AuthorizationToken string `json:"authorizationToken"`
+
+	// A reference to a Kubernetes secret containing the Dash0 authorization token. This property is optional, and is
+	// ignored if the AuthorizationToken property is set. The authorization token for your Dash0 organization
+	// can be copied from https://app.dash0.com/settings.
+	//
+	// +kubebuilder:validation:Optional
+	SecretRef string `json:"secretRef"`
+
+	// Global opt-out for self-monitoring for this operator
+	// +kubebuilder:default={enabled: true}
+	SelfMonitoring SelfMonitoring `json:"selfMonitoring,omitempty"`
+}
+
+// SelfMonitoring describes how the operator will report telemetry about its working to the
+// backend described in IngressEndpoint
+//
+// +kubebuilder:validation:Optional
+type SelfMonitoring struct {
+	// +kubebuilder:default=true
+	Enabled bool `json:"enabled"`
+}
+
+// Dash0OperatorConfigurationStatus defines the observed state of the Dash0 monitoring resource.
```

Comment:
nit replace Dash0 monitoring resource etc. throughout the file with Dash0 operator configuration resource, likewise for other spellings Dash0Monitoring dash0monitoring etc.